Nautilus/Patchers/VehicleUpgradesPatcher.cs

```diff
@@ -133,59 +171,111 @@ private static void LazyInitialize(Vehicle __instance)
 
     //
     // VEHICLE
-    // ON USE
+    // ON USE + SLOT KEY DOWN
     //
     [HarmonyPrefix]
     [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.OnUpgradeModuleUse))]
-    private static bool UpgradeModuleUseDelegate(Vehicle __instance, TechType techType, int slotID)
+    private static void UpgradeModuleUseDelegate(Vehicle __instance, TechType techType, int slotID)
     {
+        ICustomPrefab prefab;
+        UpgradeModuleGadget moduleGadget;
+        float quickSlotCharge;
+        float chargeScalar;
+        float cooldown;
         if(__instance is Exosuit)
         {
+            InternalLogger.Debug("This is an exosuit.");
             __instance = __instance as Exosuit;
-            if (!ExosuitUpgradeModules.TryGetValue(techType, out ICustomPrefab prefab))
-                return false;
+            if (!ExosuitUpgradeModules.TryGetValue(techType, out prefab))
+                return;
 
-            if (!prefab.TryGetGadget(out UpgradeModuleGadget moduleGadget))
-                return false;
+            if (!prefab.TryGetGadget(out moduleGadget))
+                return;
 
-            float quickSlotCharge = __instance.quickSlotCharge[slotID];
-            float chargeScalar = __instance.GetSlotCharge(slotID);
+            quickSlotCharge = __instance.quickSlotCharge[slotID];
+            chargeScalar = __instance.GetSlotCharge(slotID);
 
             moduleGadget.delegateOnUsed?.Invoke(__instance, slotID, quickSlotCharge, chargeScalar);
+            InternalLogger.Debug("Executed...");
 
-            var cooldown = 0f;
+            cooldown = 0f;
             if (moduleGadget.Cooldown > 0f)
                 cooldown = (float) moduleGadget.Cooldown;
 
+            InternalLogger.Debug("Cooldown set.");
+
             __instance.quickSlotTimeUsed[slotID] = Time.time;
             __instance.quickSlotCooldown[slotID] = cooldown;
 
-            return false;
+            return;
         }
-        else
-        {
+        InternalLogger.Debug("This is not an exosuit.");
 
-            if (!VehicleUpgradeModules.TryGetValue(techType, out ICustomPrefab prefab))
-                return false;
+        if (!VehicleUpgradeModules.TryGetValue(techType, out prefab))
+            return;
 
-            if (!prefab.TryGetGadget(out UpgradeModuleGadget moduleGadget))
-                return false;
+        if (!prefab.TryGetGadget(out moduleGadget))
```

Comment:
Again, couldn't this be moved to the top?

Reply:
Nope, making distinction between ExosuitUpgradeModules and VehicleUpgradeModules